src/safeposix/syscalls/net_calls.rs

```diff
@@ -1251,19 +1353,28 @@ impl Cage {
         dest_addr: &interface::GenSockaddr,
     ) -> i32 {
         //if ip and port are not specified, shunt off to send
+        //to check for a possible connection to another socket that may exist
         if dest_addr.port() == 0 && dest_addr.addr().is_unspecified() {
             return self.send_syscall(fd, buf, buflen, flags);
         }
-
+        //BUG:
+        //If fd is out of range of [0,MAXFD], process will panic
+        //Otherwise, we obtain a write gaurd to the Option<FileDescriptor> object
         let checkedfd = self.get_filedescriptor(fd).unwrap();
         let mut unlocked_fd = checkedfd.write();
+        //Check if the write gaurd holds a valid FileDescriptor
         if let Some(filedesc_enum) = &mut *unlocked_fd {
             match filedesc_enum {
+                //In this case, the file descriptor refers to a socket
                 Socket(ref mut sockfdobj) => {
+                    //Grab a write guard to the socket handle
                     let sock_tmp = sockfdobj.handle.clone();
                     let mut sockhandle = sock_tmp.write();
 
-                    // check if this is a domain socket
+                    //If the socket's domain is UNIX, return with error as UNIX
+                    //sockets are connection based.
```

Comment:
I think AF_UNIX socket could be udp (connectionless)

Reply:
Yeah, I should have specified that in my discussion with nick he explained that it happens so rarely and that there aren't really any use cases yet so we don't implement it in lind